### DISCUSSION TITLE
GitHub的「儲存庫」，就是respiratory究竟能做什麼？

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#49] [Q&A]
# 疑問
目前上課的使用，GitHub上的儲存庫=respiratory，功用共有兩個：
## 二、放置程式碼
若日後有執行程式碼，並跑出結果，可以將結果放置到GitHub的儲存庫裡嗎？GitHub本身可以讓程式動=執行程式嗎？

**Comment by JunTingLin** [comment]
GitHub並不是讓我們執行程式的地方，它的功能比較像是程式日記，每次commit一次就會被記錄下來，往後假如寫的程式爆炸了寫要倒帶，就可以回復到先前的commit紀錄，甚至假如要多人協作或是不同版號的開發還可以開branch，真的是好處多多😁

**Reply by JunTingLin** [reply]
用一張圖好像更好了解
![image](https://user-images.githubusercontent.com/92431095/157165728-009c0b40-e0fc-4661-8b5a-e2366ee78e22.png)
對吧你不會希望每次寫程是有重大版本更新都要存v1、v2、v3.....這樣吧!!
圖片節錄自: 中大程研社

**Reply by chia-kai-chang** [reply]
此外我把Github當成部署到雲端的中間倉庫。雲端空間會定期到指定的Github儲存庫柴看有沒有新的程式，如果有，就自動git pull然後重啟網站。這樣我要發布網站新功能只要git push到Github儲存庫即可，不用連到雲端主機操作。

**Reply by eason8479** [reply]
多人協作真的很方便，可以確保大家都知道目前的進度
不用改個變數就要拿usb滿世界跑，或是寄一堆mail
而且超級適合拿來抓戰犯和混分仔

**Comment by JunTingLin** [comment]
上次去參加學校社群，發現github可以透過學校信箱升級成PRO，裡面有超多免費資源😁
可以參考 [🐱‍💻申請github education有多殺!付費資源通通免費~拿到pro級標籤 #415](https://github.com/NCU-GS4719-Python/Python-Community/discussions/415)

**Comment by dabbie109303501** [comment]
GitHub的respiratory就像是程式專用的google drive，只是其他人可以透過查詢找到你公開的檔案，附帶回報錯誤或討論區的功能

**Comment by Chiway1229** [comment]
謝謝你發出這個疑問，也算是解決我心中的小問題

**Comment by ching020721** [comment]
剛接觸GitHub時，也有這樣的疑惑，但當知道這些功能時，發現真的很不錯使用，多人合作完成一個程式時github很方便
